### PR TITLE
feat: (Store) handle memory cache

### PR DESCRIPTION
**Pull request by dimamarksman**
#### Please provide a link to the associated issue.
Close: https://github.com/SAP/fundamental-ngx/issues/4843
#### Please provide a brief summary of this pull request.
This PR adds the entity server cache layer. 
It relates to cache policy strategy and base on this option wraps existing entity server service into cache service.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/store/src/lib/infrastructure/persistence/store/entity-store-builder.spec.ts [hunk 1]
```diff
@@ -31,25 +31,16 @@ class EntityCollectionServiceFactoryMock implements EntityCollectionServiceFacto
     }
 }
```
Comment (fkolar) [inline libs/store/src/lib/infrastructure/persistence/store/entity-store-builder.spec.ts:33]: I cannot comment above at the header of the file so just adding here. 

import { QueryAdapterFactory } from '../query/query-adapter'; unused import

**Review by fkolar — CHANGES_REQUESTED** [review]
I simply don't have any comment other than that unused import. I like it. I see Evgeny works is next right  to integrate.

File: libs/store/src/lib/infrastructure/persistence/store/entity-store-builder.spec.ts (hunk 1 above)
Comment (dimamarksman) [reply libs/store/src/lib/infrastructure/persistence/store/entity-store-builder.spec.ts:32]: Removed unused imports

**Review by genereg1 — APPROVED** [review]
LGTM

**Review by KevinOkamoto — APPROVED** [review]
Looks good.

**Review by fkolar — APPROVED** [review]

**Pull request merged**